### PR TITLE
add image-sha check

### PR DESCRIPTION
**Pull request by VedantMahabaleshwarkar**
Implementation for #539 

Code is pretty similar to the `latest-tag` check as the intent behind the checks are pretty similar. The `image-sha` check implements an AllowList regex instead of a blocklist that verifies if the images have an sha tag

### CODE REVIEW

**Comment by VedantMahabaleshwarkar** [comment]
I'm not sure what the test failure here is. Perhaps @janisz can tell me what I'm missing

**Comment by janisz** [comment]
You need to regenerate files

`make generated-srcs`

**Comment by janisz** [comment]
@VedantMahabaleshwarkar test is failing

**Comment by janisz** [comment]
@VedantMahabaleshwarkar ping

**Comment by janisz** [comment]
The problem is in order of tests. Bats tests should be sorted so you need to move your test after `hpa-minimum-three-replicas`.

I'll think how to make error message better as I also needed some time to understand what's wrong. Maybe before making diff we should check order.

**Comment by VedantMahabaleshwarkar** [comment]
@janisz Thanks for the help! It worked, all tests are passing now

**Comment by VedantMahabaleshwarkar** [comment]
@janisz pinging again for a review on this PR

**Comment by janisz** [comment]
I'm on PTO.  I'll take a look next week.

**Comment by janisz** [comment]
@VedantMahabaleshwarkar Please regenerate files

**Comment by VedantMahabaleshwarkar** [comment]
@janisz hey sorry for the delay. I changed the test to accept `all` filetypes instead of just `deploymentlike` and I have updated the docs as well. I think the tests are passing now.

**Comment by VedantMahabaleshwarkar** [comment]
@janisz Can this PR be merged?

File: pkg/builtinchecks/yamls/image-sha.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: "image-sha"
+description: "Indicates if there are any image references that are not references by sha256 tags"
+remediation: "Reference all images using their sha256 tags."
+scope:
+  objectKinds:
+    - "Any"
+template: "image-sha"
```
Comment (janisz) [inline pkg/builtinchecks/yamls/image-sha.yaml:7]:
```suggestion
template: "latest-tag"
```

Comment (janisz) [reply pkg/builtinchecks/yamls/image-sha.yaml:7]: We could reuse the latest-tag template. probably we should change it's name

**Review by janisz — COMMENTED** [review]
Please reuse existing latest-tag template and not copy it.

```diff
- pkg/templates/imageshatag/internal/params/gen-params.go
- pkg/templates/imageshatag/internal/params/params.go
- pkg/templates/imageshatag/template.go
- pkg/templates/imageshatag/template_test.go

diff --git a/e2etests/bats-tests.sh b/e2etests/bats-tests.sh
index 37e0491..d074400 100755
--- a/e2etests/bats-tests.sh
+++ b/e2etests/bats-tests.sh
@@ -359,8 +359,8 @@ get_value_from() {
   message2=$(get_value_from "${lines[0]}" '.Reports[1].Object.K8sObject.GroupVersionKind.Kind + ": " + .Reports[1].Diagnostic.Message')
   count=$(get_value_from "${lines[0]}" '.Reports | length')
 
-  [[ "${message1}" == "Deployment: The container \"app\" is using an invalid container image, \"app:v1\". Please reference the image using a SHA256 tag." ]]
-  [[ "${message2}" == "DeploymentConfig: The container \"app\" is using an invalid container image, \"app:v1\". Please reference the image using a SHA256 tag." ]]
+  [[ "${message1}" == 'Deployment: The container "app" is using an invalid container image, "app:v1". Please use images that satisfies the `AllowList` criteria : [".*:[a-fA-F0-9]{64}$"]' ]]
+  [[ "${message2}" == 'DeploymentConfig: The container "app" is using an invalid container image, "app:v1". Please use images that satisfies the `AllowList` criteria : [".*:[a-fA-F0-9]{64}$"]' ]]
   [[ "${count}" == "2" ]]  
 }
 
diff --git a/pkg/builtinchecks/yamls/image-sha.yaml b/pkg/builtinchecks/yamls/image-sha.yaml
index dd5eeb8..3aa99a5 100644
--- a/pkg/builtinchecks/yamls/image-sha.yaml
+++ b/pkg/builtinchecks/yamls/image-sha.yaml
@@ -4,6 +4,6 @@ remediation: "Reference all images using their sha256 tags."
 scope:
   objectKinds:
     - "Any"
-template: "image-sha"
+template: "latest-tag"
 params:
   AllowList: [".*:[a-fA-F0-9]{64}$"] 
diff --git a/pkg/templates/all/all.go b/pkg/templates/all/all.go
index 9e0056d..db88467 100644
--- a/pkg/templates/all/all.go
+++ b/pkg/templates/all/all.go
@@ -24,7 +24,6 @@ import (
        _ "golang.stackrox.io/kube-linter/pkg/templates/hostpid"
        _ "golang.stackrox.io/kube-linter/pkg/templates/hpareplicas"
        _ "golang.stackrox.io/kube-linter/pkg/templates/imagepullpolicy"
-       _ "golang.stackrox.io/kube-linter/pkg/templates/imageshatag"
        _ "golang.stackrox.io/kube-linter/pkg/templates/latesttag"
        _ "golang.stackrox.io/kube-linter/pkg/templates/livenessprobe"
        _ "golang.stackrox.io/kube-linter/pkg/templates/memoryrequirements"
```

**Review by janisz — COMMENTED** [review]
ping
